src/components/dashboard/EntityStatOverview/useMonthlyUsage.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery } from '@supabase-cache-helpers/postgrest-swr';
+import { DefaultStatsWithDocument, getStatsForDashboard } from 'api/stats';
+import { DateTime } from 'luxon';
+import { useMemo } from 'react';
+import { useTenantStore } from 'stores/Tenant/Store';
+import { CatalogStats_Dashboard } from 'types';
+import { hasLength } from 'utils/misc-utils';
+
+// The interfaces of this union type have minimal overlap and a type guard is required
+// to access the properties specific to one of the interfaces.
+const isDefaultStatistic = (
+    datum: CatalogStats_Dashboard | DefaultStatsWithDocument
+): datum is DefaultStatsWithDocument => 'bytes_read_by_me' in datum;
+
+export default function useMonthlyUsage() {
+    const selectedTenant = useTenantStore((state) => state.selectedTenant);
+    const endDate = DateTime.utc().startOf('month');
+
+    const query = useMemo(
+        () =>
+            hasLength(selectedTenant)
+                ? getStatsForDashboard(selectedTenant, 'monthly', endDate)
+                : null,
+        [endDate, selectedTenant]
+    );
+
```

Comment:
I'm not sure... cause linting isn't complaining... but I should have thought that `endDate` would cause this memo to always rerun.

Reply:
Also - noting our convo from Slack. Looks like the values are being cached and I think we should yank that out for now.